### ISSUE TITLE
Warnings and invalid HTML generated from `gutenberg_render_elements_support()` for empty block

### ISSUE THREAD

**Issue by bobbingwide** [WordPress/gutenberg#33401]
## Description
After updating to Gutenberg 11.0.0 I noticed a number of Warnings being produced.
I analysed the problem in https://github.com/bobbingwide/thisis/issues/34
and eventually determined that the problem occurs when the block being rendered has produced no output
but has a `style` attribute for the link color.

- For my block this happened because the user was not logged in.
- For the scenario to reproduce the problem below it occurs because the block is not registered.
- The filter function is still being run against blank output. 
- This causes the Warnings to be displayed and the invalid output is generated.

## Step-by-step reproduction instructions
1. Add a new post.
2. Enter Code edit
3. Paste this content. 

```
<!-- wp:paragraph --><p>Apart from this paragraph the only content in the post is another block with a style attribute. The template only displays the post-content. What could possibly go wrong?</p><!-- /wp:paragraph -->
<!-- wp:reproduce/issue34 {"style":{"elements":{"link":{"color":{"text":"var:preset|color|dark-grey"}}}}} /-->
```
4. Save the post.
5. View the post on the front end. See the unexpected output. 
6. Re edit the post
7. Change `style` to `smyle`
8. Save and view again
9. The problem goes away.

## Expected behaviour
The block should not appear on the front end. 
There shouldn't be any attempt to generate a `<style>` tag. 

## Actual behaviour
Warning messages and invalid HTML generated
```
Warning: Undefined array key 0 in C:\apache\htdocs\wp55\wp-content\plugins\gutenberg\lib\block-supports\elements.php on line 51

Warning: Trying to access array offset on value of type null in C:\apache\htdocs\wp55\wp-content\plugins\gutenberg\lib\block-supports\elements.php on line 51

Warning: Undefined array key 0 in C:\apache\htdocs\wp55\wp-content\plugins\gutenberg\lib\block-supports\elements.php on line 63

Warning: Trying to access array offset on value of type null in C:\apache\htdocs\wp55\wp-content\plugins\gutenberg\lib\block-supports\elements.php on line 63

Apart from this paragraph the only content in the post is another block with a style attribute. The template only displays the post-content. What could possibly go wrong?

class=”wp-elements-60edf9307a0c3″
```

## Screenshots or screen recording (optional)
This is what I got:

![image](https://user-images.githubusercontent.com/2474435/125517327-e2d6b749-1cc0-421b-862a-3177dbbda122.png)

This is what @annezazu got
<img width="1360" alt="Screen Shot 2021-07-13 at 2 04 49 PM" src="https://user-images.githubusercontent.com/2474435/125520612-7a68b9e4-7446-49ec-acd9-f3d094f9d8bd.png">

## Code snippet (optional)

Proposed fix:
Return early when the block content is empty

```
function gutenberg_render_elements_support( $block_content, $block )
{
    if ( empty( $block_content ) ) {
        return $block_content;
    }
``` 

## WordPress information
- WordPress version:  5.7.2 | 5.8-RC2
- Gutenberg version: 11.0.0 
- Are all plugins except Gutenberg deactivated? No ( @bobbingwide ) 
- Are you using a default theme (e.g. Twenty Twenty-One)? No ( @bobbingwide )

## Device information
For @bobbingwide
- Device: Desktop 
- Operating system: Windows 10 
- Browser:

**Comment by Mamaduka** [comment]
Hi, @bobbingwide

I believe this was resolved via #35836.

**Comment by talldan** [comment]
Sounds like it is resolved, so I'll close this.
